app/src/main/java/com/keylesspalace/tusky/components/systemnotifications/NotificationService.kt

```diff
@@ -0,0 +1,791 @@
+     *   - No need to especially track the grouping
+     *   - No need to change an existing single notification when there arrives another one of its group
+     *   - Only the summary one will get announced
+     */
+    private fun createSummaryNotification(account: AccountEntity, type: Notification.Type, additionalNotifications: List<Notification>): NotificationWithIdAndTag? {
+        val typeChannelId = getChannelId(account, type) ?: return null
+
+        val summaryStackBuilder = TaskStackBuilder.create(context)
+        summaryStackBuilder.addParentStack(MainActivity::class.java)
+        val summaryResultIntent = openNotificationIntent(context, account.id, type)
+        summaryStackBuilder.addNextIntent(summaryResultIntent)
+
+        val summaryResultPendingIntent = summaryStackBuilder.getPendingIntent(
+            (notificationId + account.id * 10000).toInt(),
+            pendingIntentFlags(false)
+        )
+
+        val activeNotifications = getActiveNotifications(account.id, typeChannelId)
+
+        val notificationCount = activeNotifications.size + additionalNotifications.size
+
+        val title = context.resources.getQuantityString(R.plurals.notification_title_summary, notificationCount, notificationCount)
+        val text = joinNames(activeNotifications, additionalNotifications)
+
+        val summaryBuilder = NotificationCompat.Builder(context, typeChannelId)
+            .setSmallIcon(R.drawable.ic_notify)
+            .setContentIntent(summaryResultPendingIntent)
+            .setColor(context.getColor(R.color.notification_color))
+            .setAutoCancel(true)
+            .setContentTitle(title)
+            .setContentText(text)
+            .setShortcutId(account.id.toString())
+            .setSubText(account.fullName)
+            .setVisibility(NotificationCompat.VISIBILITY_PRIVATE)
+            .setCategory(NotificationCompat.CATEGORY_SOCIAL)
+            .setGroup(typeChannelId)
+            .setGroupSummary(true)
+            .setGroupAlertBehavior(NotificationCompat.GROUP_ALERT_SUMMARY)
+
+        setSoundVibrationLight(account, summaryBuilder)
+
+        val summaryTag = "$GROUP_SUMMARY_TAG.$typeChannelId"
+
+        return NotificationWithIdAndTag(summaryTag, account.id.toInt(), summaryBuilder.build())
+    }
+
+    fun createWorkerNotification(@StringRes titleResource: Int): android.app.Notification {
+        val title = context.getString(titleResource)
+        return NotificationCompat.Builder(context, CHANNEL_BACKGROUND_TASKS)
+            .setContentTitle(title)
+            .setTicker(title)
+            .setSmallIcon(R.drawable.ic_notify)
+            .setOngoing(true)
+            .build()
+    }
+
+    private fun getChannelId(account: AccountEntity, type: Notification.Type): String? {
+        return when (type) {
+            Notification.Type.MENTION -> CHANNEL_MENTION + account.identifier
```

Comment:
Why is `CHANNEL_MENTION` a constant here and the others below aren't? I don't care either way it should just be consistent 😅

Reply:
It's the only one that is used from somewhere else.